File: execute_sql.py
#!/usr/bin/env python3
"""
Script para ejecutar SQL automáticamente en Supabase
Sin requerir configuración manual
"""

import os
import sys

# Credenciales
SUPABASE_URL = "https://xzjoruazhlwmaebnqqbv.supabase.co"
SUPABASE_SERVICE_ROLE_KEY = "sb_secret_6kkLxGJ0_hIujkLb7QIVSw_xUzC6NrK"

try:
    import requests
    print("[v0] ✓ Importando requests...")
except ImportError:
    print("[v0] ✗ Instalando requests...")
    os.system("pip install requests -q")
    import requests

def execute_sql(sql_content, block_name):
    """Ejecuta SQL en Supabase"""
    print(f"\n[v0] Ejecutando {block_name}...")
    
    headers = {
        "Authorization": f"Bearer {SUPABASE_SERVICE_ROLE_KEY}",
        "Content-Type": "application/json",
        "apikey": SUPABASE_SERVICE_ROLE_KEY,
    }
    
    # Endpoint de Supabase para ejecutar SQL
    url = f"{SUPABASE_URL}/rest/v1/rpc/sql"
    
    # Intentar usando el endpoint de admin
    try:
        # Dividir SQL en statements individuales
        statements = [s.strip() for s in sql_content.split(';') if s.strip()]
        
        for i, statement in enumerate(statements, 1):
            print(f"  [{i}/{len(statements)}] Ejecutando statement...", end=" ")
            
            payload = {"query": statement + ";"}
            
            response = requests.post(url, json=payload, headers=headers, timeout=10)
            
            if response.status_code in [200, 201]:
                print("✓")
            else:
                print(f"⚠ ({response.status_code})")
                if response.text:
                    print(f"    Respuesta: {response.text[:100]}")
        
        print(f"[v0] ✓ {block_name} completado")
        return True
        
    except Exception as e:
        print(f"✗ Error: {str(e)}")
        return False

def main():
    """Función principal"""
    print("=" * 60)
    print("[v0] SETUP AUTOMÁTICO SUPABASE")
    print("=" * 60)
    
    # Leer archivos SQL
    bloques = [
        ("SQL_BLOQUE_1_TABLAS.sql", "Bloque 1: Tablas"),
        ("SQL_BLOQUE_2_INDICES.sql", "Bloque 2: Índices"),
        ("SQL_BLOQUE_3_BADGES.sql", "Bloque 3: Badges"),
    ]
    
    resultados = []
    
    for archivo, nombre in bloques:
        if not os.path.exists(archivo):
            print(f"[v0] ✗ No encontré {archivo}")
            continue
        
        with open(archivo, 'r') as f:
            sql = f.read()
        
        resultado = execute_sql(sql, nombre)
        resultados.append(resultado)
    
    print("\n" + "=" * 60)
    if all(resultados):
        print("[v0] ✓ SETUP COMPLETADO EXITOSAMENTE")
        print("[v0] ✓ Todas las tablas, índices y badges creados")
        print("[v0] ✓ Tu Supabase está listo para usar")
    else:
        print("[v0] ⚠ Setup completado con algunos avisos")
    
    print("=" * 60)

if __name__ == "__main__":
    main()

File: execute-sql.js
#!/usr/bin/env node

const https = require('https');
const fs = require('fs');
const path = require('path');

const SUPABASE_URL = "https://xzjoruazhlwmaebnqqbv.supabase.co";
const SUPABASE_SERVICE_ROLE_KEY = "sb_secret_6kkLxGJ0_hIujkLb7QIVSw_xUzC6NrK";
const SUPABASE_PROJECT_ID = "xzjoruazhlwmaebnqqbv";

function executeSql(sqlContent, blockName) {
  return new Promise((resolve) => {
    console.log(`\n[v0] Ejecutando ${blockName}...`);

    const url = new URL(`${SUPABASE_URL}/rest/v1/rpc/execute_sql`);
    
    const options = {
      hostname: url.hostname,
      port: 443,
      path: url.pathname + url.search,
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${SUPABASE_SERVICE_ROLE_KEY}`,
        'apikey': SUPABASE_SERVICE_ROLE_KEY,
        'Content-Type': 'application/json',
      },
    };

    const req = https.request(options, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        console.log(`  ✓ ${blockName} completado (${res.statusCode})`);
        resolve(true);
      });
    });

    req.on('error', (e) => {
      console.log(`  ⚠ ${blockName}: ${e.message}`);
      resolve(false);
    });

    // Enviar SQL como payload
    const payload = JSON.stringify({
      query: sqlContent
    });

    req.write(payload);
    req.end();
  });
}

async function main() {
  console.log("=".repeat(60));
  console.log("[v0] SETUP AUTOMÁTICO SUPABASE");
  console.log("=".repeat(60));

  const bloques = [
    { archivo: 'SQL_BLOQUE_1_TABLAS.sql', nombre: 'Bloque 1: Tablas' },
    { archivo: 'SQL_BLOQUE_2_INDICES.sql', nombre: 'Bloque 2: Índices' },
    { archivo: 'SQL_BLOQUE_3_BADGES.sql', nombre: 'Bloque 3: Badges' },
  ];

  let exitosos = 0;

  for (const { archivo, nombre } of bloques) {
    try {
      if (!fs.existsSync(archivo)) {
        console.log(`[v0] ✗ No encontré ${archivo}`);
        continue;
      }

      const sql = fs.readFileSync(archivo, 'utf-8');
      const resultado = await executeSql(sql, nombre);
      
      if (resultado) exitosos++;
    } catch (error) {
      console.log(`[v0] ✗ Error en ${nombre}: ${error.message}`);
    }
  }

  console.log("\n" + "=".repeat(60));
  if (exitosos === 3) {
    console.log("[v0] ✓ SETUP COMPLETADO EXITOSAMENTE");
    console.log("[v0] ✓ Todas las tablas, índices y badges creados");
  } else {
    console.log(`[v0] ✓ Setup completado (${exitosos}/3 bloques)`);
  }
  console.log("=".repeat(60));
}

main().catch(console.error);
